### DISCUSSION TITLE
The future of PIE Example Apps

### DISCUSSION THREAD

**Discussion by siggerzz** [justeattakeaway/pie#2031] [General]
Recently we deleted some example apps in the repo, but after looking at the remaining ones, it looks like all of them only implement pie-button, despite all importing the pie-webc package.

I think it's pretty clear looking at git history these example apps aren't getting maintained because there's currently no enforcement to update them during PR stage, and the last commit for all example apps was 4 months ago to implement `pie-webc`.

Could we come to consensus on what to do with the remaining ones, because currently, it seems like a waste of CI compute / time / AWS maintenance in its current state.

Off the top of my head, the options that come to mind are:

- Delete them and don't migrate them to PIE Aperture

- Delete them and migrate to PIE Aperture, and prioritise tickets to make sure they're aligned.

- Keep them in PIE and prioritise tickets to make sure they're aligned PIE Aperture, and add some Danger / CI enforcement to ensure they're maintained.

To add to this, I think we also need some Danger / CI enforcement to ensure PIE Aperture is getting updated. It would seem silly adding such checks to maintain the PIE example apps when we have nothing in place to Aperture, when ultimately, those apps are more important.

Aperture hasn't been touched in three weeks, primarily because the pie-cookie-banner wasn't tested prior to merging and [caused the apps to break](https://github.com/justeattakeaway/pie-aperture/pull/223). We also can't merge the PR to promote cookie-banner to 1.x because the vanilla-app not longer works as expected with the new version of cookie banner; more info can be found in DSW-2562

**Comment by ashleynolan** [comment]
In my head, the main reason left for them was as documentation for users of certain frameworks when implementing PIE.

I spoke to @jamieomaguire about when we were deleting some of the other example apps, and I'm happy deleting these, if we replace them with codesandbox style demo docs showing how to implement a component using PIE.

I'm pretty sure Jamie did this for at least one application – we probably need to make a ticket to go through the others so we delete the remaining applications.

**Comment by ashleynolan** [comment]
To have the Aperture discussion in a separate thread – what check should we have? Is there a way of enforcing this outside of having checkboxes for people to test in Aperture (which were removed in the above example I think)
